assignment6.py

```diff
@@ -0,0 +1,23 @@
+def swap(l, begin, end):
+    tmp = l[begin]
+	l[begin] = l[end]
+	l[end] = tmp
+
+def rearrage(src, target):
+
+	free = 0
+
+	src_ind = dict([(a, i) for i, a in enumerate(src)])
+	target_ind = dict([(a, i) for i, a in enumerate(target)])
+
+	i = src_ind[free]
```

Comment:
Here you are using free as index, might be that the name is a bit misleading to me, but index 0 is not necessarily the index of the free space (ie. I might have a car there).
So I don't understand how then the if condition follow